components/eamxx/tests/coupled/dynamics_physics/homme_mam4xx_pg2/CMakeLists.txt

```diff
@@ -72,25 +72,16 @@ foreach (file IN ITEMS ${TEST_INPUT_FILES})
   GetInputFile(${file})
 endforeach()
 
+# Compare output files produced by npX tests, to ensure they are bfb
+include (CompareNCFilesFamily)
 
-## Finally compare all MPI rank output files against the single rank output as a baseline, using CPRNC
-## Only if running with 2+ ranks configurations
-# This test requires CPRNC
-if (TEST_RANK_END GREATER TEST_RANK_START)
-  include (BuildCprnc)
-  BuildCprnc()
+CreateRange(MpiRanks ${TEST_RANK_START} ${TEST_RANK_END})
 
-  math (EXPR CMP_RANK_START ${TEST_RANK_START}+1)
-  foreach (MPI_RANKS RANGE ${CMP_RANK_START} ${TEST_RANK_END})
-
-    set (SRC_FILE "${BASE_TEST_NAME}_output.INSTANT.nsteps_x${NUM_STEPS}.np${MPI_RANKS}.${RUN_T0}.nc")
-    set (TGT_FILE "${BASE_TEST_NAME}_output.INSTANT.nsteps_x${NUM_STEPS}.np${TEST_RANK_START}.${RUN_T0}.nc")
-    set (TEST_NAME "${BASE_TEST_NAME}_np${TEST_RANK_START}_vs_np${MPI_RANKS}_output_bfb")
-    add_test (NAME ${TEST_NAME}
-              COMMAND cmake -P ${CMAKE_BINARY_DIR}/bin/CprncTest.cmake ${SRC_FILE} ${TGT_FILE}
-              WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
-    set_tests_properties(${TEST_NAME} PROPERTIES
-      LABELS "dynamics;physics;mam4xx"
-      FIXTURES_REQUIRED "${FIXTURES_BASE_NAME}_np${MPI_RANKS}_omp1;${FIXTURES_BASE_NAME}_np1_omp1")
-  endforeach()
-endif()
+CompareNCFilesFamily (
```

Comment:
I was pondering whether I should add a higher level macro, called something like `CompareNCFilesFamilyPEM`, which takes `RANK_BEG`, `RANK_END`, and `RANK_INC` and internally creates the list of MAGIC_VALUES to then fwd to a call to `CompareNCFilesFamily`. It would not make the code _much_ smaller, but it would allow to not have to manually call `CreateRange(...)`.

Let me know if you think it would make sense to add it.

Reply:
Yeah, I think it makes sense to remove the manual call to `CreateRange(...)`. Makes the tests CMakeList clearer/simpler/easier when it comes to adding new tests. Not a big deal though if you decide you want to leave it out.